src/app/fdctl/main1.c

```diff
@@ -170,6 +172,14 @@ fdctl_boot( int *        pargc,
       strncpy( config->log.path, log_path, sizeof( config->log.path ) - 1 );
   }
 
+  char * args[ MAX_ARGC+2 ];
+  for( int i=0; i<*pargc; i++ ) args[ i ] = *( *pargv+i );
+  /* Override --shmem-path value from the config */
+  args[ *pargc   ] = "--shmem-path";
```

Comment:
--shmem-path is the only possible argument to fd_shmem_private_boot, so you don't need to copy args here. You can just pass a 3 element array of `["--shmem-path", path, NULL]`